### DISCUSSION TITLE
Return None instead of raising an exception

### DISCUSSION THREAD

**Discussion by jeaboswell** [python-gitlab/python-gitlab#1734] [Q&A] [answered]
I am working with this module, and I thought I had gotten it to return None when making requests for objects (users, projects, etc.) that don't exist using urllib3.disable_warnings(). I discovered today, however, that it is throwing 404 errors. Does anyone know of a way to return None, instead of raising exceptions?

**Comment by JohnVillalovos** [comment, marked as answer]
I would think to use a try/except block and catch the exception. It should raise a Gitlab exception.

A quick guess of how it would look.

```python
try:
    user = gl.users.get(INVALID_ID)
except gitlab.exceptions.GitlabGetError:
   user = None
```

**Reply by jeaboswell** [reply]
Okay. This was the solution I had in mind. I was almost certain disabling warnings with urllib3 got it to return None, but no matter what I do today, I can't get the same result.
